src/libexpr/flake/flake.cc

```diff
@@ -373,6 +373,29 @@ LockedFlake lockFlake(
         {
             debug("computing lock file node '%s'", printInputPath(inputPathPrefix));
 
+            auto overrides2 = overrides;
+            for (auto & [inputPath, inputOverride] : overrides2) {
+                auto inputPath2(inputPath);
+                auto follow = inputPath2.back();
+                inputPath2.pop_back();
+                if (inputPath2 == inputPathPrefix
+                    && flakeInputs.find(follow) == flakeInputs.end()
+                ) {
+                    std::string root;
+                    for (auto & element : inputPath2) {
+                        root.append(element);
+                        if (element != inputPath2.back()) {
+                            root.append(".inputs.");
+                        }
+                    }
+                    throw Error(
```

Comment:
I think a warning would be better. There's probably some somewhat legitimate scenarios where you want to override something that doesn't exist yet (say because you're generating some overrides on a set of flakes or because you're using `--override-input` to point an input to a new copy which has less inputs itself) so I wouldn't want it to just fail

Reply:
Fine by me. I'm happy as long as there's an indication that something might be wrong :)